cmd/releasego/publish-announcement.go

```diff
@@ -200,23 +201,36 @@ func publishAnnouncement(p subcmd.ParseFunc) (err error) {
 		return fmt.Errorf("file %s already exists in go-devblog repository", blogFilePath)
 	}
 
+	// Create a feature branch (gitpr convention: dev/<purpose>/<name>) and open a PR to main.
+	prSet := gitpr.PRRefSet{Name: releaseInfo.Slug, Purpose: "blog"}
+	branchName := prSet.PRBranch()
+
 	if err := githubutil.Retry(func() error {
-		// Upload the announcement to the go-devblog repositoryy main branch with proper commit message
 		if err := githubutil.UploadFile(
 			ctx,
 			client,
 			org,
 			repo,
-			"main",
+			branchName,
 			blogFilePath,
-			fmt.Sprintf("Add new blog post for new release in %s", releaseDate.Format("2006-01-02")),
+			fmt.Sprintf("Add blog post: %s", releaseInfo.Title),
 			content.Bytes()); err != nil {
-			return fmt.Errorf("error uploading file to go-devblog repository : %w", err)
+			return fmt.Errorf("error uploading file to branch %s: %w", branchName, err)
 		}
-
 		return nil
 	}); err != nil {
-		return fmt.Errorf("error checking if file exists in go-devblog repository : %w", err)
+		return err
+	}
+
+	// Create PR using gitpr.
+	auther, err := gitHubAuthFlags.NewAuther()
+	if err != nil {
+		return fmt.Errorf("failed to get GitHub auther: %w", err)
+	}
+	ownerRepo := fmt.Sprintf("%s/%s", org, repo)
+	prReq := prSet.CreateGitHubPR(org, releaseInfo.Title, "Automated PR: add Microsoft Go release announcement.")
+	if _, err := gitpr.PostGitHub(ownerRepo, prReq, auther); err != nil && !errors.Is(err, gitpr.ErrPRAlreadyExists) {
```

Comment:
Missing enabling auto-merge for the new PR?

Reply:
I'm testing it.

Reply:
> often want to add release notes

This is extremely rare in my experience. Also:

* This is too late. It shouldn't be up to the release runner to put them together on the release day, the infra should feed from what the dev responsible for the change has written in https://github.com/microsoft/go/tree/microsoft/main/docs (or somewhere similar) and has been reviewed by the team over there. Perhaps we need a new location if we want to e.g. summarize some CVE changes and need to prepare those notes privately.
* Submitting a PR to change the file is no more (or perhaps actually less, given the GitHub UI!) effort than changing an existing PR.

Reply:
You haven't enabled automerge. You'll also want to have the review bot approve the PR